### PR TITLE
refactor: remove code redundancy

### PR DESCRIPTION
**Pull request by mDemianchuk**

### CODE REVIEW

File: src/main/java/com/kaniademianchuk/model/Manager.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
     private final Map<Integer, T> allDevices;
 
     public Manager() {
-        this.allDevices = new HashMap<>();
+        this(new HashMap<>());
```
Comment (NickDemianchuk) [inline src/main/java/com/kaniademianchuk/model/Manager.java]: code reuse yo
meh, but it reduces readability too

File: src/main/java/com/kaniademianchuk/model/Manager.java [hunk 2]
```diff
@@ -36,10 +36,9 @@ public boolean removeDevice(T device) {
     }
 
     public boolean removeDevice(Integer id) {
-        if (!this.allDevices.containsKey(id)) {
+        if(this.allDevices.remove(id) == null) {
```
Comment (NickDemianchuk) [inline src/main/java/com/kaniademianchuk/model/Manager.java]: reduced readability, but shorter code
can change it back

File: src/main/java/com/kaniademianchuk/model/Scheduler.java [hunk 3]
```diff
@@ -1,7 +1,5 @@
 package com.kaniademianchuk.model;
 
-import com.kaniademianchuk.api.ITogglable;
```
Comment (NickDemianchuk) [inline src/main/java/com/kaniademianchuk/model/Scheduler.java]: unused import

File: src/main/java/com/kaniademianchuk/model/Manager.java (hunk 2 above)
Comment (RobusK) [reply src/main/java/com/kaniademianchuk/model/Manager.java]: Return remove == null

No need for if

File: src/main/java/com/kaniademianchuk/model/TogglableGroup.java [hunk 4]
```diff
@@ -24,23 +24,25 @@ protected TogglableGroup(Integer id, String name, T... devices) {
 
     @Override
     public void turnOn() {
```
Comment (NickDemianchuk) [inline src/main/java/com/kaniademianchuk/model/TogglableGroup.java]: don't see a reason to cast

Comment (RobusK) [reply src/main/java/com/kaniademianchuk/model/TogglableGroup.java]: Reason for what

File: src/main/java/com/kaniademianchuk/model/Manager.java (hunk 2 above)
Comment (NickDemianchuk) [reply src/main/java/com/kaniademianchuk/model/Manager.java]: can't do
return type is Optional not null
thus, Optional.empty() == null -> false

File: src/main/java/com/kaniademianchuk/model/TogglableGroup.java (hunk 4 above)
Comment (NickDemianchuk) [reply src/main/java/com/kaniademianchuk/model/TogglableGroup.java]: edited my comment

**Pull request merged**